proposals/bindless.md

```diff
@@ -199,106 +199,126 @@ Thi means that **Bindings cannot be overwritten before `onSubmittedWorkDone` sin
+    readonly attribute unsigned long maxResourceTableSize;
 };
 ```
 
-#### Bindgroup and layout creation
+#### Resource tables creation
 
-Dynamic binding arrays are added to `GPUBindGroups` as a second, variable-size, sparse and (totally or partially) heterogeneous range of bindings positioned after the regular WebGPU bindings.
-The `GPUBindGroupLayoutDescriptor` gains a `GPUDynamicBindingArrayLayout` member that describes where the range of bindings for the dynamic binding array starts, and what it may contain.
-Likewise `GPUBindGroupDescriptor` gains a `dynamicArraySize` that describes the actual size of the dynamic binding array.
-Because `GPUBindGroup` become objects with variable-sized contents, they get a new `destroy()` method to free the GPU memory without waiting on the GC.
+The `GPUResourceTable` creation only takes the size of the resource table as an argument.
+Because the `GPUResourceTable` is an object containing data, it gets a `.destroy()` method as well as a reflection of its creation parameters.
 
 ```webidl
-enum GPUDynamicBindingType {
-    "sampled-texture",
-    "storage-texture",
-    "storage-buffer",
-    "texel-buffer",
-    "sampler",
-};
-
-dictionary GPUDynamicBindingArrayLayout {
-    unsigned long start = 0;
-    GPUDynamicBindingType type;
-};
-
-partial dictionary GPUBindGroupLayoutDescriptor {
-    GPUDynamicBindingArrayLayout dynamicArray;
+dictionary GPUResourceTableDescriptor : GPUObjectDescriptorBase {
+    required unsigned long size;
 };
+partial interface GPUResourceTable {
+    void destroy();
 
-partial dictionay GPUBindGroupDescriptor {
-    unsigned long dynamicArraySize;
+    readonly unsigned long size;
 };
-
-partial interface GPUBindGroup {
-    void destroy();
+partial interface GPUDevice {
+    GPUResourceTable createResourceTable(GPUResourceTableDescriptor descriptor);
 };
 ```
 
-`GPUDynamicBindingArrayLayout.type` chooses what kind of resources the dynamic binding array my contain.
-When `heterogenous-dynamic-binding-array` is enabled, it may be set to `undefined` in which case the dynamic binding array may contain any binding compatible with an other value of `GPUDynamicBindingType`.
-The size of the dynamic binding array is not known when creating the layout, so it uses all the bindings in `[GPUDynamicBindingArrayLayout.start, Infinity)`.
-The use of a dynamic binding array implicitly uses a storage buffer binding for all stages for the `GPUBindGroupLayout` (this is required for the implementation of the validation of bindless).
+The steps for `GPUDevice.createResourceTable(desc)` are:
 
-Validation rules added to `GPUDevice.createBindGroupLayout(desc)` when `desc.dynamicArray` is specified:
+ - Let `t` be a new WebGPU object (this, GPUResourceTable, desc).
+ - let `t.size` be `Math.min(desc.size, this.limits.maxResourceTableSize)`. (Note: this is done to avoid the need to create giant content-timeline arrays if `size` is huge but fails validation on the device timeline)
```

Comment:
nit: Other resources expose their originally-requested size even if it's too big. I'd say do that, and internally just don't allocate those arrays in that case.